### PR TITLE
Configure firewall ports

### PR DESCRIPTION
**Pull request by tahsinrahman**

### CODE REVIEW

**Review by dibosh — APPROVED** [review]
LGTM

File: main.tf [hunk 1]
```diff
@@ -41,6 +40,10 @@ locals {
         secret_name    = secret_name
         secret_data    = secret_data
   }]])
+
+  firewall_name  = format("%s-%s", "ingress-allow", var.name_suffix)
+  istio_ports    = ["10250", "443", "15017", "8080", "15000"]
```
Comment (dibosh) [inline main.tf:45]: Let's break this line down like what roman has done in another PR. Basically showing which ports have what significance.

Comment (tahsinrahman) [reply main.tf]: done

File: main.tf [hunk 2]
```diff
@@ -41,6 +40,13 @@ locals {
         secret_name    = secret_name
         secret_data    = secret_data
   }]])
+
+  firewall_name = format("%s-%s", "ingress-allow", var.name_suffix)
+  istio_ports = [
+    "10250", "443", "15017", # see https://istio.io/latest/docs/setup/platform-setup/gke/
+    "8080", "15000",         # see https://kiali.io/documentation/latest/installation-guide/#_google_cloud_private_cluster_requirements
+  ]
+  firewall_ports = toset(concat(var.additional_firewall_ports, length(toset(var.istio_ip_names)) > 0 ? local.istio_ports : []))
```
Comment (syedrakib) [inline main.tf:49]:
```suggestion
  all_ingress_ports = toset(concat(var.additional_firewall_ports, length(toset(var.istio_ip_names)) > 0 ? local.istio_ports : []))
```

File: variables.tf [hunk 3]
```diff
@@ -59,6 +59,12 @@ variable "istioctl_firewall_name" {
   default     = "allow-istioctl"
 }
 
+variable "additional_firewall_ports" {
```
Comment (syedrakib) [inline variables.tf:62]:
```suggestion
variable "additional_ingress_ports" {
```

**Review by syedrakib — CHANGES_REQUESTED** [review]

File: main.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (tahsinrahman) [reply main.tf]: there are two things to take care here, one is istio is enabled, second is additional_ingress_port is set

[the remark this answers is not in the document]
Comment (syedrakib) [reply main.tf]: That's ok. If istio is enabled, then only we populate `local.istio_ports`. Otherwise, `local.istio_ports` can be `[]`. So that decision is taken care of in the context `local.istio_ports` itself.

Here, we just concat it. 

Hence, If `local.istio_ports` is empty, then here it will be empty + var.additional_firewall_ports. If `local.istio_ports` has values, then here it will be some values + var.additional_firewall_ports

File: main.tf [hunk 4]
```diff
@@ -41,6 +40,13 @@ locals {
         secret_name    = secret_name
         secret_data    = secret_data
   }]])
+
+  firewall_name = format("%s-%s", var.firewall_name, var.name_suffix)
+  istio_ports = [
```
Comment (syedrakib) [inline main.tf:45]:
```suggestion
  istio_ports = length(toset(var.istio_ip_names)) == 0 ? [] : [ 
```

File: main.tf [hunk 5]
```diff
@@ -41,6 +40,13 @@ locals {
...
+    "10250", "443", "15017", # see https://istio.io/latest/docs/setup/platform-setup/gke/
+    "8080", "15000",         # see https://kiali.io/documentation/latest/installation-guide/#_google_cloud_private_cluster_requirements
+  ]
+  all_ingress_ports = toset(concat(var.additional_ingress_ports, length(toset(var.istio_ip_names)) > 0 ? local.istio_ports : []))
```
Comment (syedrakib) [inline main.tf:49]:
```suggestion
  all_ingress_ports = toset(concat(var.additional_ingress_ports, local.istio_ports))
```

File: variables.tf [hunk 6]
```diff
@@ -53,10 +53,16 @@ variable "istio_ip_names" {
   default     = []
 }
 
-variable "istioctl_firewall_name" {
-  description = "An arbitrary name to identify the IstioCtl firewall that will be generated for the GKE cluster if \"var.istio_ip_names\" contains any values."
+variable "firewall_name" {
+  description = "An arbitrary name to identify the firewall that will be generated for the GKE cluster if \"var.istio_ip_names\" or \"additional_ingress_ports\" contains any values."
```
Comment (syedrakib) [inline variables.tf:57]:
```suggestion
  description = "An arbitrary name to identify the firewall that will be generated for the GKE cluster if \"var.istio_ip_names\" or \"var.additional_ingress_ports\" contains any values."
```

**Review by syedrakib — APPROVED** [review]
